src/spec.rs

```diff
@@ -0,0 +1,176 @@
+//! The contents of this module require the rustgc feature. To avoid polluting
```

Comment:
Maybe silly question: why is the file called `spec`?

Reply:
A I missed this one. It's just short for specialization/specializer. Perhaps its not a great name, but allocation_specialization felt a bit long winded.

Reply:
Ah, I always think "spec" means "specification". I think we need a different filename, but I don't have a firm opinion on what that might be :)

Reply:
How about just `specialization`?

Reply:
I'm fine with that. If you want something simpler, you could call it something like `picker` maybe?

Reply:
Let's go with `specialization`, I think if we went for `picker` we should rename the structs in the file for consistency.

Reply:
I'm happy to go with whatever you decide on this one.

Reply:
Fixed 812827a440560dfd6d845cecd3953fe4009fac16